guardrails/run/async_stream_runner.py

```diff
@@ -275,6 +284,9 @@ def get_chunk_text(self, chunk: Any, api: Union[PromptCallableBase, None]) -> st
         """Get the text from a chunk."""
         chunk_text = ""
 
+        if not chunk.choices or len(chunk.choices) == 0:
+            return chunk_text
+
```

Comment:
The try-except blocks below this line have a fallback for situations when `chunk` has come from a `generator of strings`; this check does not have such a fallback.

Also, the `ValueError` in this method can never be raised: it is raised only if intepreter is unable to do the `chunk_text = chunk` assignment.

```python
if isinstance(chunk, str):
    finished = False
    content = chunk
elif hasattr(chunk, "choices"):
    if not chunk.choices or len(chunk.choices) == 0:
        return ""
    choice = chunk.choices[0]
    finished = choice.finish_reason
    content = getattr(choice, "text", getattr(choice, "delta").content)
else:
    raise ValueError("...")

if finished:
    return ""
return content
```

Reply:
Thanks for the peer-review.  I've added a few more safeguards as well and made the ValueErrors both reachable and more descriptive.